stage #2.5/module #4 REST Basics - Spring Web/REST Basics/content.md

```diff
@@ -0,0 +1,526 @@
+# REST Basics - Spring Web
+
+## Basics of RESTful APIs
+_**REST is an API architecture style.**_ It stands for Representational State Transfer (REST). 
+RREST is an architectural style that defines a set of constraints to be used for creating web services. 
+REST API is a way of accessing web services in a simple and flexible way without having any processing.
+REST technology is generally preferred to the more robust Simple Object Access Protocol (SOAP) technology because REST uses less bandwidth, 
+simple and flexible making it more suitable for internet usage. It’s used to fetch or give some information from a web service. 
+All communication done via REST API uses only HTTP request.
+
+## Richardson REST API Maturity Model
+The Richardson REST Maturity Model describes four different levels of REST (starting at Level 0). A REST API that supports hypermedia controls is classified as Level 3 in this maturity model.<br>
+- **Level 0 - a single URI and use a single HTTP method (typically POST):** <br> These services have a single URI and use a single HTTP method (typically POST). These are the most primitive way of building SOA applications with a single POST method and using XML to communicate between services.Level zero of maturity does not make use of any of URI, HTTP Methods, and HATEOAS capabilities.
+
+- **Level 1 - different URLs for different resources but use a single HTTP method (typically POST):** <br> API design at Level 1 is all about using different URLs to interact with the different resources in your application but only a single HTTP verb – generally HTTP POST. These services will give each resource, available in the application, a unique URI. A unique URI separately identifies one unique resource – and that makes these REST API better than level zero.
```

Comment:
Please use everywhere URl, not URL
It's confusing when you switch from one to another